apps/backend/src/resource/event/event.service.ts

```diff
@@ -295,11 +308,31 @@ export class EventService {
             });
     }
 
-    async delete(eventId: number) {
-        await this.eventRepository.delete({ id: eventId });
+    async delete(userId: number, eventId: number) {
+        return await this.eventRepository
+            .findOneByOrFail({ id: eventId, createdById: userId })
+            .then(async () => {
+                await this.eventRepository.delete({ id: eventId });
+            })
+            .catch(() => {
+                throw new NotFoundException('Event not found');
+            });
```

Comment:
I think it would be better to do it separately. So first find and then delete. And I wonder if it is a good place for transaction